test/templates/enums/.jhipster/A.json

```
@@ -0,0 +1,32 @@
+{
+    "name": "A",
+    "fields": [
+        {
+            "fieldName": "myFieldA",
+            "fieldType": "MyEnumA",
+            "fieldValues": "AAA,BBB"
+        },
+        {
+            "fieldName": "myFieldB",
+            "fieldType": "MyEnumB",
+            "fieldValues": "AAA (aaa_aaa),BBB"
```

I think we should change to 
```suggestion
            "fieldValues": "AAA (aaa aaa),BBB"
```
White space here should be almost done.

---

JCore and the generator don't produce this

---

The generator will with the 2 changes I am proposing to getEnumValuesWithCustomValues.
And 1 of the 2 must be the first to support, preferably the generator.

---

I will not block the PR due to this change.
Once the integration are added and i18n is fixed, the PR is done.
Integration test is more important than mocha test here.
But I will create another PR in sequence to add white space support.

---

So you know, the feature was intended as JDL-only. Are you sure you want to introduce it in the generator? I don't mind but I'd rather do less work than more :)

---

Why JDL only? Nothing should be JDL only in my opinion.
Once it’s supported on generator-jhipster, it can by added by blueprints and is really easy to add a prompt.

---

I disagree. Some things are just too ill-suited to do in CLIs. If I recollect this correctly this was the reason and I still agree with this. As the JDL is a "visible" thing, it's easy to do such a thing as having a custom enum label. In the CLI, you have to know how to do it prior to writing your values (that are unformatted). Or at least you have to inform the user of the possibility. Which can bloat the output

---

As I said your PR, your call.
But jhipster is a platform, not jdl dependent (which is lagging behind without blueprint support).
More then that, we are a community, if someone needs some feature that will not interfere too much on the maintenance, I see no reason to decline.
Anyway, you should add enums to integration test and fix i18n.
Currently the main IT tests are not jdl only.
So we can merge this PR.